sdk/react/panels/cast/CastDevicesScreen.js

```diff
@@ -0,0 +1,198 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Animated,
+  ListView,
+  Text,
+  View,
+  TouchableHighlight,
+} from 'react-native';
+
+import {
+  BUTTON_NAMES,
+} from '../../constants';
+import Utils from '../../utils';
+
+import castScreenStyles from '../style/castScreenStyles.json';
+
+const styles = Utils.getStyles(castScreenStyles);
+
+let ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 === r2 });
+
+const dismissButtonSize = 20;
+const castButtonSize = 35;
+
+class CastDevicesScreen extends Component {
+  static propTypes = {
+    height: PropTypes.number.isRequired,
+    width: PropTypes.number.isRequired,
+    onDismiss: PropTypes.func.isRequired,
+    onDeviceSelected: PropTypes.func.isRequired,
+    config: PropTypes.object.isRequired,
+    deviceIds: PropTypes.array.isRequired,
+    deviceNames: PropTypes.array.isRequired,
+    selectedItem: PropTypes.string.isRequired,
+  };
+
+  componentWillMount() {
+    const { height, deviceNames } = this.props;
+
+    this.state = {
+      translateY: new Animated.Value(height),
+      opacity: new Animated.Value(0),
+      dataSource: ds.cloneWithRows(deviceNames),
+      selectedID: -1,
+    };
+  }
+
+  componentDidMount() {
+    const { height } = this.props;
+    const { translateY, opacity } = this.state;
+
+    translateY.setValue(height);
+    opacity.setValue(0);
+
+    Animated.parallel([
+      Animated.timing(
+        translateY,
+        {
+          toValue: 0,
+          duration: 700,
+          delay: 0,
+        },
+      ),
+      Animated.timing(
+        opacity,
+        {
+          toValue: 1,
+          duration: 500,
+          delay: 0,
+        },
+      ),
+    ])
+      .start();
+  }
+
+  onDismissBtnPress = () => {
+    const { onDismiss } = this.props;
+    onDismiss();
+  };
+
+  onDismissPress = () => {
+    const { opacity } = this.state;
+    Animated.timing(
+      opacity,
+      {
+        toValue: 0,
+        duration: 500,
+        delay: 0,
+      },
+    )
+      .start(this.onDismissBtnPress);
+  };
+
+  onPressButton(rowID) {
+    const { dataSource, translateY, opacity } = this.state;
+    const { deviceNames, onDeviceSelected, deviceIds } = this.props;
+
+    this.setState({
+      selectedID: rowID,
+      dataSource: dataSource.cloneWithRows(deviceNames),
```

Comment:
do you expect that `deviceNames` will be different at this moment?

Reply:
fixed

Reply:
deleted as it doesn't used at all